plutus-contract/src/Plutus/Contract/Test/ContractModel.hs

```diff
@@ -1134,3 +1143,35 @@ checkNoLockedFundsProof options spec NoLockedFundsProof{nlfpMainStrategy   = mai
                               "  " ++ show bal'
                     DL.assert err (bal `leq` bal')
 
+
+-- | Check that running a contract model does not result in validation
+-- failures due to partial functions from the prelude being used in
+-- a validator.
+checkNoPartiality :: ContractModel m
+                  => [ContractInstanceSpec m]
+                  -> Property
+checkNoPartiality = checkNoPartialityWithOptions defaultCheckOptions
+
+-- | Check that running a contract model does not result in validation
+-- failures due to partial functions from the prelude being used in
+-- a validator.
+checkNoPartialityWithOptions :: forall m. ContractModel m
+                             => CheckOptions
+                             -> [ContractInstanceSpec m]
+                             -> Property
+checkNoPartialityWithOptions opts handleSpecs = forAllShrink arbitrary shrink $ \ acts -> go check acts
+  where
+    check = checkOnchain .&&. (assertNoFailedTransactions .||. checkOffchain)
+    checkOnchain = assertChainEvents checkEvents
+
+    checkOffchain = assertFailedTransaction (\ _ _ -> all (either checkEvent (const True) . sveResult))
+
+    checkEvent (EvaluationError [] "CekEvaluationFailure")  = False
```

Comment:
Hmph. It's a bit sad that we only have the string representation here. I guess we could embed the actual error type in there...